File: 643-maximum-average-subarray-i/maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:

        l = 0
        r = k
        wSum : float= 0
        mSum : float =float('-inf')
        for i in range(l,r):
            wSum = wSum + nums[i]
        mSum = wSum
        while r < len(nums):
            
           
            wSum -= nums[l]
            wSum +=nums[r]
            
            l+=1
            r+=1
            if wSum/k > mSum/k:
                mSum = wSum
        return mSum/k


        

        

File: 643-maximum-average-subarray-i/maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int MaxWindowSum = 0;
        int wsum = 0;
        
        for(int i = 0; i<k;i++){
            wsum = wsum + nums[i];
        }

        MaxWindowSum = wsum;

        for(int i = k ; i < nums.length;i++){
            wsum = wsum - nums[i-k] + nums[i];
            if (wsum > MaxWindowSum) {
                MaxWindowSum = wsum;
            }   
        }
    
        return (double) MaxWindowSum / k;       
    }
}